src/Raven.Server/Documents/Queries/IndexQueryServerSide.cs

```diff
@@ -275,6 +296,9 @@ private static void AssertPaging(IndexQueryServerSide indexQuery)
             if (indexQuery.Limit < 0)
                 throw new InvalidQueryException($"{nameof(Limit)} ({nameof(PageSize)}) cannot be negative, but was {indexQuery.Limit}.", indexQuery.Query, indexQuery.QueryParameters);
 
+            if (indexQuery.ScanLimit < 0)
+                throw new InvalidQueryException($"{nameof(ScanLimit)} cannot be negative, but was {indexQuery.ScanLimit}.", indexQuery.Query, indexQuery.QueryParameters);
```

Comment:
@ayende Can it be 0? In Client API @maciejaszyk has a check `<= 0`

Reply:
Wouldn't be meanful, I think. If you have a scan limit with a filter, you always return 0

Reply:
> Wouldn't be meanful, I think. If you have a scan limit with a filter, you always return 0

When scan_limit is 0, right?